x/gov/keeper/vote.go

```diff
@@ -58,6 +58,23 @@ func (keeper Keeper) AddVote(ctx context.Context, proposalID uint64, voterAddr s
 
 // deleteVotes deletes the all votes from a given proposalID.
 func (keeper Keeper) deleteVotes(ctx context.Context, proposalID uint64) error {
-	// TODO(tip): fix https://github.com/cosmos/cosmos-sdk/issues/16162
+	rng := collections.NewPrefixedPairRange[uint64, sdk.AccAddress](proposalID)
+	iter, err := keeper.Votes.Iterate(ctx, rng)
```

Comment:
I think it's more readable if we use the `Clear` API: https://github.com/cosmos/cosmos-sdk/blob/main/collections/map.go#L157